test/jdk/java/net/httpclient/http2/server/Http2TestServerConnection.java

```diff
@@ -770,6 +772,16 @@ void readLoop() {
                                 // but the continuation, even after a reset
                                 // should be handle gracefully by the client
                                 // anyway.
+                            } else if ((stream & 0x01) == 0x01 && stream < next) {
+                                // We may receive a reset on a client stream that has already
+                                // been closed. Just ignore it.
+                                System.err.println("TestServer: received ResetFrame on closed stream: " + stream);
+                                System.err.println(frame);
+                            } else if ((stream & 0x01) == 0x00 && stream < nextPush) {
+                                // We may receive a reset on a push stream that has already
+                                // been closed. Just ignore it.
+                                System.err.println("TestServer: received ResetFrame on closed push stream: " + stream);
+                                System.err.println(frame);
                             } else {
```

Comment:
For clarity, I'd suggest adding static methods for testing if a streamid is client or server.
Otherwise, the change looks fine.

Reply:
Thanks Michael. Done!